File: src/main/java/corp/soyummy/ingredients/Ingredient.java
package corp.soyummy.ingredients;

import lombok.Data;
import lombok.Builder;

import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Field;

@Data
@Builder
@Document(collection = "ingredients")
public class Ingredient {
    @Id
    private String id;

    @Field("ttl")
    private String title;

    @Field("desc")
    private String description;

    @Field("thb")
    private String thumbnail;
}

File: src/main/kotlin/com/soyummy/ingredients/Ingredient.kt
package com.soyummy.ingredients

import org.springframework.data.mongodb.core.mapping.Document
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Field
import org.springframework.data.mongodb.core.index.Indexed
import org.springframework.data.mongodb.core.index.TextIndexed

@Document(collection = "ingredients")
data class Ingredient(
    @Id
    val id: String? = null,

    @Field("ttl")
    @Indexed(unique = true)
    @TextIndexed(weight = 3f)
    val title: String,

    @Field("desc")
    @TextIndexed(weight = 1f)
    val description: String,

    @Field("thb")
    val thumbnail: String
)